dask/array/tests/test_array_core.py

```diff
@@ -2121,6 +2121,18 @@ def test_point_slicing():
     result = d.vindex[[1, 2, 5, 5], [3, 1, 6, 1]]
     assert_eq(result, x[[1, 2, 5, 5], [3, 1, 6, 1]])
 
+    result = d.vindex[da.asarray([1, 2, 6, 6]), :]
+    assert_eq(result, x[[1, 2, 6, 6], :])
+
+    result = d.vindex[:, da.asarray([1, 2, 7, 7])]
+    assert_eq(result, x[:, [1, 2, 7, 7]])
+
+    result = d.vindex[2, da.asarray([1, 2, 7, 7])]
+    assert_eq(result, x[2, [1, 2, 7, 7]])
+
+    result = d.vindex[(d % 3).nonzero()]
+    assert_eq(result, x[(x % 3).nonzero()])
```

Comment:
It would be reassuring to see more cases where the values of the array are 

1.  out of order
2.  chunked differently.
3.  negative

Reply:
Sorry, could you please explain what you mean by "out of order"?

Reply:
da.asarray([7, 7, 1, 6])

Reply:
Have added a couple explicit tests like this. Interestingly the `nonzero` case also captured these issues too. Though explicit test cases seem preferable.